src/osd/PrimaryLogPG.cc

```diff
@@ -727,6 +728,35 @@ void PrimaryLogPG::maybe_force_recovery()
     maybe_kick_recovery(soid);
 }
 
+class PGLSNameFilter : public PGLSFilter {
+  boost::regex regex;
+public:
+  int init(bufferlist::iterator &params) override
+  {
+    try {
+      string reg_exp ;
+      ::decode(reg_exp, params);
+      if (reg_exp.empty()) {
+        return -EINVAL;
+      } else {
+        regex.assign(reg_exp, boost::regex_constants::no_except);
+      }
+    } catch (buffer::error &e) {
+      return -EINVAL;
+    }
+    return 0;
+  }
+  bool filter(const hobject_t &obj, bufferlist& indata,
+                            bufferlist& outdata) override;
```

Comment:
Nit: indentation spacing ... and since the function is really only one line long, why not just inline the definition here it while you're at it?